pkg/library/storage/commonStorage.go

```diff
@@ -80,34 +87,56 @@ func RewriteContainerVolumeMounts(workspaceId string, podAdditions *v1alpha1.Pod
 		return err
 	}
 
-	podAdditions.Volumes = append(podAdditions.Volumes, corev1.Volume{
-		Name: commonPVCName,
-		VolumeSource: corev1.VolumeSource{
-			PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
-				ClaimName: commonPVCName,
+	for _, component := range ephemeralVolumes {
+		vol := corev1.Volume{
+			Name: component.Name,
+			VolumeSource: corev1.VolumeSource{
+				EmptyDir: &corev1.EmptyDirVolumeSource{},
+			},
+		}
+		if component.Volume.Size != "" {
+			sizeResource, err := resource.ParseQuantity(component.Volume.Size)
+			if err != nil {
+				return fmt.Errorf("failed to parse size for Volume %s: %w", component.Name, err)
+			}
+			vol.EmptyDir.SizeLimit = &sizeResource
+		}
+		podAdditions.Volumes = append(podAdditions.Volumes, vol)
+	}
+
+	if NeedsStorage(workspace) {
+		podAdditions.Volumes = append(podAdditions.Volumes, corev1.Volume{
```

Comment:
Do we need to do L67-88 if workspace does not need a storage? Maybe it makes sense to extend this condition body?

Reply:
Technically no, but we do also check for invalid volume references there (`volume, ok := devfileVolumes[vm.Name]`). However, it probably makes sense to separate out that check since at some point we'll just use some validation library from devfile/api to not have to worry about it.

Reply:
Reworked this section